acme/acme/client_test.py

```diff
@@ -127,6 +127,13 @@ def test_query_registration(self):
         self.response.json.return_value = self.regr.body.to_json()
         self.assertEqual(self.regr, self.client.query_registration(self.regr))
 
+    def test_query_registration_updates_new_authzr_uri(self):
```

Comment:
Query registration passes in the previous value and the existing code does not update the link (i.e. it ignores a new value sent by the server and uses the current one if it is not None).  Unless of course the current value in the regr object is None (which would then be the right thing to change for this test).